pandas/core/nanops.py

```diff
@@ -637,7 +619,7 @@ def _maybe_null_out(result, axis, mask):
             else:
                 result = result.astype('f8')
             result[null_mask] = np.nan
-    else:
+    elif result is not tslib.NaT:
```

Comment:
if we started with ``M8/m8`` and then do a ``.view('i8')`` this needs to be compared to ``pd.lib.iNaT``, not sure why you are not hitting this here